src/Sylius/Bundle/ProductBundle/Controller/ProductAttributeController.php

```diff
@@ -89,14 +95,36 @@ protected function getAttributeFormsInAllLocales(AttributeInterface $attribute,
         $attributeForm = $this->get('sylius.form_registry.attribute_type')->get($attribute->getType(), 'default');
 
         $forms = [];
+
+        if (!$attribute->isTranslatable()) {
+            array_push($localeCodes, null);
+
+            return $this->createFormAndView($attributeForm, $attribute, null, $forms);
+        }
+
         foreach ($localeCodes as $localeCode) {
-            $forms[$localeCode] = $this
-                ->get('form.factory')
-                ->createNamed('value', $attributeForm, null, ['label' => $attribute->getName(), 'configuration' => $attribute->getConfiguration()])
-                ->createView()
-            ;
+            $forms = $this->createFormAndView($attributeForm, $attribute, $localeCode, $forms);
```

Comment:
```suggestion
            $forms[$localeCode] = $this->createFormAndView($attributeForm, $attribute, $localeCode);
```
and properly adjust the private method, for me it would be a more clear approach than a method that overwrites an entire array in foreach.